src/Catalog/Persistence/AzureStorage.cs

```diff
@@ -96,7 +105,7 @@ public override bool Exists(string fileName)
 
         protected override async Task<StorageContent> OnLoad(Uri resourceUri, CancellationToken cancellationToken)
         {
-            string name = GetName(resourceUri);
+            string name = GetName(resourceUri).TrimStart('/');
```

Comment:
add a comment to why this cleanup is necessary, and why it can't be done by GetName proper